server_http.hpp

```diff
@@ -80,6 +80,7 @@ namespace SimpleWeb {
             std::string remote_endpoint_address;
             unsigned short remote_endpoint_port;
             
+            std::function<void(const std::exception&)> exception_handler;
```

Comment:
I guess we could change it to Request::address() (return ip adress) and Request::port() (return port) too, and let the user of the library handle any exceptions (though I can't see any cases where these functions can/should throw). Guess we should add noexcept on all the methods that does not throw at the same time. What do you think?

And yes, one point is to hide boost::asio from the user of the library. It's a pretty difficult library, and most people do not need to learn it.